collect_app/src/main/java/org/odk/collect/android/entities/JsonFileEntitiesRepository.kt

```diff
@@ -77,27 +90,37 @@ class JsonFileEntitiesRepository(directory: File) : EntitiesRepository {
         StrictMode.noteSlowCall("Reading from JSON file")
 
         if (!entitiesFile.exists()) {
-            entitiesFile.parentFile.mkdirs()
-            entitiesFile.createNewFile()
+            createFile()
         }
 
-        val typeToken = TypeToken.getParameterized(
-            MutableMap::class.java,
-            String::class.java,
-            TypeToken.getParameterized(MutableList::class.java, JsonEntity::class.java).type
-        )
+        try {
+            val typeToken = TypeToken.getParameterized(
+                MutableMap::class.java,
+                String::class.java,
+                TypeToken.getParameterized(MutableList::class.java, JsonEntity::class.java).type
+            )
 
-        val json = entitiesFile.readText()
-        return if (json.isNotBlank()) {
-            val parsedJson =
-                Gson().fromJson<MutableMap<String, MutableList<JsonEntity>>>(json, typeToken.type)
+            val json = entitiesFile.readText()
+            return if (json.isNotBlank()) {
+                val parsedJson =
+                    Gson().fromJson<MutableMap<String, MutableList<JsonEntity>>>(json, typeToken.type)
 
-            parsedJson
-        } else {
-            mutableMapOf()
+                parsedJson
+            } else {
+                mutableMapOf()
+            }
+        } catch (e: Exception) {
+            entitiesFile.delete()
```

Comment:
I think you could have a separate method like:
```java
private fun deleteFile() {
         entitiesFile.delete()
         entitiesFile.createNewFile()
    }
```
You don't have to call #createFile as the parent directory should still exist right? so you just need to create the file again.